### PR TITLE
Dissalow execution of is_write cards in dashboards, embedding and pulses

### PR DESCRIPTION
**Pull request by escherize**
Addresses #22847

Throws a 405 (not allowed) when trying to execute a card that has is_write set to true through outside of the actions api.

- [x] Cards
  - [x] POST `api/card/:card-id/query`
  - [x] POST `api/card/:card-id/query/:export-format`
  - [x] POST `api/card/pivot/:card-id/query`

- [x] Dashboards
  - [x] POST `api/dashboard/:dashboard-id/dashcard/:dashcard-id/card/:card-id/query`
  - [x] POST `api/dashboard/:dashboard-id/dashcard/:dashcard-id/card/:card-id/query/:export-format`
  - [x] POST `api/dashboard/pivot/:dashboard-id/dashcard/:dashcard-id/card/:card-id/query`

- [x] Embedding
  - [x] GET `api/preview_embed/dashboard/:token/dashcard/:dashcard-id/card/:card-id`
  - [x] GET `api/preview_embed/card/:token/query`
  - [x] GET `api/preview_embed/pivot/card/:token/query`
  - [x] GET `api/public/dashboard/:uuid/dashcard/:dashcard-id/card/:card-id`
  - [x] GET `api/public/pivot/dashboard/:uuid/dashcard/:dashcard-id/card/:card-id`

- [x] Pulses
  - [x] execute-dashboard (called when running pulses)
  - [x] POST `api/pulse/test` (Test send an unsaved pulse)

### CODE REVIEW

File: src/metabase/api/common.clj [hunk 1]
```diff
@@ -413,6 +413,13 @@
     (check (not (:archived object))
       [404 {:message (tru "The object has been archived."), :error_code "archived"}])))
 
+(defn check-is-readonly
+  "Check that the object has `:is_write` = false, or throw a `405`. Returns `object` as-is if check passes."
+  [object]
+  (u/prog1 object
+    (check (not (:is_write object))
+      [405 {:message (tru "Write queries are only executable via the Actions API."), :error_code "is_not_readonly"}])))
```
Comment (camsaul) [inline src/metabase/api/common.clj:421]: I think 405s make sense vs 400s for this but I'd double check with the FE folks that it's not going to break anything just to be sure

**Review by camsaul — COMMENTED** [review]
Can we get some tests for this? Besides that it looks good to me

**Review by camsaul — CHANGES_REQUESTED** [review]
Please add some tests 🙏

File: src/metabase/api/common.clj (hunk 1 above)
Comment (escherize) [reply src/metabase/api/common.clj:428]: Good point, I checked with them and they're not using `x == 400`.

**Comment by escherize** [comment]
This branch had some conflicts, I fixed them and opened this PR: #24712

**Pull request closed without merging**